controllers/spifilecontentrequest_controller.go

```diff
@@ -74,28 +95,18 @@ func (r *SPIFileContentRequestReconciler) SetupWithManager(mgr ctrl.Manager) err
 	if err := r.finalizers.Register(linkedFileRequestBindingsFinalizerName, &linkedFileRequestBindingsFinalizer{client: r.K8sClient}); err != nil {
 		return fmt.Errorf("failed to register the linked bindings finalizer: %w", err)
 	}
-
 	err := ctrl.NewControllerManagedBy(mgr).
 		For(&api.SPIFileContentRequest{}).
 		Watches(&source.Kind{Type: &api.SPIAccessTokenBinding{}}, handler.EnqueueRequestsFromMapFunc(func(o client.Object) []reconcile.Request {
-			fileRequests := &api.SPIFileContentRequestList{}
-			if err := r.K8sClient.List(context.Background(), fileRequests, client.InNamespace(o.GetNamespace())); err != nil {
-				spiFileContentRequestLog.Error(err, "Unable to fetch file content requests list", "namespace", o.GetNamespace())
-				return []reconcile.Request{}
-			}
-			ret := make([]reconcile.Request, 0, len(fileRequests.Items))
-			for _, fr := range fileRequests.Items {
-				if fr.Status.LinkedBindingName == o.GetName() {
-					ret = append(ret, reconcile.Request{
-						NamespacedName: types.NamespacedName{
-							Name:      fr.Name,
-							Namespace: fr.Namespace,
-						},
-					})
-				}
-			}
+			ret := make([]reconcile.Request, 0, 1)
```

Comment:
I don't understand why this construct? why first make 0 slice with capped size to 1 then immediatelly append and return? why not:
```
return []reconcile.Request{
reconcile.Request{
				NamespacedName: types.NamespacedName{
					Name:      o.GetLabels()[LinkedFileRequestLabel],
					Namespace: o.GetNamespace(),
				},
			}
}
```
?